src/observer/index.js

```diff
@@ -172,11 +173,27 @@ function copyAugment (target, src, keys) {
 
 function defineReactive (obj, key, val) {
   var dep = new Dep()
-  var childOb = Observer.create(val)
+
+  var target = {
+    val: val
+  }
+
+  if (config.convertAllProperties) {
+    var property = Object.getOwnPropertyDescriptor(obj, key)
+    if (property && property.get && property.set) {
```

Comment:
Just noticed it's possible for a property to only have getters, or only have setters. We should handle all three possible situations here. Probably need to procedurally build up the descriptor based on the presence of the getter and the setter.